dev/WindowsAppRuntime_BootstrapDLL/MddBootstrapAutoInitializer.cpp

```diff
@@ -22,15 +23,13 @@ namespace Microsoft::Windows::ApplicationModel::DynamicDependency::Bootstrap
 
         static void Initialize()
         {
-            const UINT32 c_majorMinorVersion{ WINDOWSAPPSDK_RELEASE_MAJORMINOR };
-            PCWSTR c_versionTag{ WINDOWSAPPSDK_RELEASE_VERSION_TAG_W };
-            const PACKAGE_VERSION c_minVersion{ WINDOWSAPPSDK_RUNTIME_VERSION_UINT64 };
-            const HRESULT hr{ ::MddBootstrapInitialize(c_majorMinorVersion, c_versionTag, c_minVersion) };
+            const PACKAGE_VERSION c_minVersion{WINDOWSAPPSDK_RUNTIME_VERSION_UINT64};
+            const HRESULT hr{::MddBootstrapInitialize(WINDOWSAPPSDK_RELEASE_MAJORMINOR, WINDOWSAPPSDK_RELEASE_VERSION_TAG_W, c_minVersion)};
             if (FAILED(hr))
             {
-                exit(hr);
+                __fastfail(hr);
```

Comment:
[Docs](https://docs.microsoft.com/cpp/intrinsics/fastfail?view=msvc-170) say the parameter needs to be unsigned int (HRESULT is LONG) and a FAST_FAIL_<description> constant

```
code
[in] A FAST_FAIL_<description> symbolic constant from winnt.h or wdm.h that indicates the reason for process termination.
```

Is any old HRESULT legal here?

Reply:
Can this instead be:

```c++
FAIL_FAST_IF_FAILED(::MddBootstrapInitialize(...));
```

Reply:
other than a dependence on wil.h, what does that gain?

#define FAIL_FAST_HR(hr) {__fastfail(hr);}
#define FAIL_FAST_IF_FAILED(x) { HRESULT _hr = (x); if (FAILED(_hr)) { FAIL_FAST_HR(_hr); } }

Reply:
Ahhh. Nevermind... ;-)